### PR TITLE
default port tcp

### PR DESCRIPTION
**Pull request by chayim**
closes issue #98

### CODE REVIEW

**Review by jugmac00 — CHANGES_REQUESTED** [review]
Please have a look at the broken test suite.

File: tox_docker/__init__.py [hunk 1]
```diff
@@ -183,16 +183,15 @@ def getenvdict(reader, key):
 
 
 def _validate_port(port_line):
-    host_port, _, container_port_proto = port_line.partition(":")
-    host_port = int(host_port)
-
-    container_port, _, protocol = container_port_proto.partition("/")
-    container_port = int(container_port)
+    try:
+        host_port, container_port, protocol = re.findall(reg, port_line)[0]
```
[the remark this answers is not in the document]
Comment (chayim) [reply tox_docker/__init__.py:188]: Pushed again - sorry about that.

**Comment by jugmac00** [comment]
I removed myself as a reviewer as I neither use nor maintain this package. I just commented on the broken build to speed up the review process for the core maintainers.

File: README.rst [hunk 2]
```diff
@@ -59,8 +59,9 @@ The ``[docker:container-name]`` section may contain the following directives:
 ``ports``
     A multi-line list of port mapping specifications, as
     ``HOST_PORT:CONTAINER_PORT/PROTO``, which causes the container's
-    ``EXPOSE`` d port to be available on ``HOST_PORT``. See below for
-    more on port mapping.
+    ``EXPOSE`` d port to be available on ``HOST_PORT``. In the case where
+    the protocol is not specified, tox-docker assumes tcp, as per the
+    dockre default. See below for more on port mapping.
```
Comment (gaborbernat) [inline README.rst:64]: typo here

**Review by gaborbernat — CHANGES_REQUESTED** [review]
We should also add some tests 👍

**Comment by dcrosta** [comment]
I'm leaning towards not accepting this -- the inconvenience of having to type "tcp:" seems a small and infrequent price to pay for greater explicitness in the tox-docker config. Does anyone feel really strongly about this?

**Comment by chayim** [comment]
Doing this is aligns tox-docker more closelt with docker. My bias is to
standardize on the behaviour of the tool with which I'm integrating. I
expected tox-docker to require port declarations to match the docker
syntax, it helps reduce the barrier to entry.

**Comment by gaborbernat** [comment]
I'm personally leaning towards @chayim here, but not strongly. Also note the main maintainer here is @dcrosta so ultimately it's up to his decision.

**Comment by chayim** [comment]
> We should also add some tests

@gaborbernat Happy to add the tests and finish this off. But if @dcrosta doesn't want this behaviour then we should kill it off.

**Comment by chayim** [comment]
Ping @dcrosta do you have a preference - prior to my writing the tests?

**Comment by chayim** [comment]
Given the lack of answer - I assume we don't want this. Closing.

**Pull request closed without merging**